### PR TITLE
Fixes not being able to throw stuff into disposals

### LINKED ISSUE #9718: [BUG] Can't throw anything into disposal bins anymore.
### There is no other issue reporting the same problem.

- [X] I have searched the existing issues

### Game Version

23031117

### I was

Client

### OS

Windows 8, 10, 11

### What happened?

Throwing objects at a disposal bin no longer has them go in at all, make a sound, or say anything about whether it bounced off or made it in.

### What should have happened instead?

Throwing an object at a disposal bin should have it make a sound, make it say something in chat if you made it in or not, and be able to go in.

### Steps to reproduce the issue

1. Throw object at disposal bin

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

 CL: [Fix] Fixes not being able to throw stuff into disposals

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs [hunk 1]
```diff
@@ -1519,7 +1519,7 @@ public void FlyingUpdateMe()
 					{
 						if (isServer)
 						{
-							if ((UniversalObjectPhysics) bump == this) continue;
+							if (bump as UniversalObjectPhysics == this) continue;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs:1522]: why is this different?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs:1522]: it does some automatic null check in the background i believe

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs:1522]: ah just remembered, the first one will throw, the second doesn't throw but leaves the value as null

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs:1522]: apparently throws an error if it can't cast with the other one

**Review by corp-0 — APPROVED** [review]

**Pull request merged**